.github/workflows/deploy.yml

```diff
@@ -85,10 +119,52 @@ jobs:
     steps:
     - name: Clone repository
       uses: actions/checkout@v4
-    - name: Download and extract artifacts
+    - name: Download and extract built artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: build.artifacts
+    - name: Download and extract reused linux_arm32 artifacts
+      if: ${{ needs.plan_builds.outputs.download_linux_arm32_runid }}
+      uses: actions/download-artifact@v4
+      with:
+        path: build.artifacts
+        run-id: ${{ needs.plan_builds.outputs.download_linux_arm32_runid }}
+        github-token: ${{ github.token }}
```

Comment:
You should be able to select a specific artifacts [like this](https://github.com/actions/download-artifact?tab=readme-ov-file#download-multiple-filtered-artifacts-to-the-same-directory):

```YAML
with:
  pattern: MU4_*_Lin_armv7l_*
```

It gets tricky for the non-ARM Linux. This might work if more advanced [shell-like globbing](https://www.man7.org/linux/man-pages/man7/glob.7.html) is supported:

```YAML
with:
  pattern: MU4_*_Lin_[0-9]*
```

Or simply:

```YAML
with:
  pattern: MU4_*_Lin_4*
```